modules/blockbuilder/blockbuilder.go

```diff
@@ -409,6 +406,39 @@ outer:
 	return lastRec.Timestamp, offset.At, nil
 }
 
+func (b *BlockBuilder) commitOffset(ctx context.Context, offset kadm.Offset, group string, partition int32) error {
+	offsets := make(kadm.Offsets)
+	offsets.Add(offset)
+
+	boff := backoff.New(ctx, backoff.Config{
+		MinBackoff: 100 * time.Millisecond,
+		MaxBackoff: time.Minute,
+		MaxRetries: 10,
+	})
+	for boff.Ongoing() {
+		resp, err := b.kadm.CommitOffsets(ctx, group, offsets)
+		commitOffsetErr := resp.Error()
+		if err == nil && commitOffsetErr == nil {
+			break
+		}
+		if !kerr.IsRetriable(commitOffsetErr) {
```

Comment:
Do we need to see if `err` is retriable, and log `commitOffsetErr`?  Finding the dual error handling a little unclear.

Reply:
`CommitAllOffsets`, yep looks better 👍